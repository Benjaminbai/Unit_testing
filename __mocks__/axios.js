
const mockAxios = jest.requireActual('Axios')

export default mockAxios;