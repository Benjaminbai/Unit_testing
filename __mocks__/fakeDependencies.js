jest.mock('react-router-dom', () => {
    const originalModule = jest.requireActual('react-router-dom');

    return {
        __esModule: true,
        ...originalModule,
        useLocation: jest.fn().mockReturnValue({
            pathname: '',
            search: '',
            hash: '',
            state: null,
            key: '',
        }),
        useHistory: () => ({
            push: jest.fn(),
            location: {}
        }),
        Redirect: jest.fn().mockReturnValue({}),
    }
})
window.matchMedia = jest.fn().mockImplementation((query) => {
    const obj = {
        matches: false,
        media: query,
        onchange: null,
        addListener: jest.fn(),
        removeListener: jest.fn(),
    };

    return obj;
})

jest.mock('react', () => ({
    ...jest.requireActual('react'),
    useLayoutEffect: jest.requireActual('react').useEffect
}))