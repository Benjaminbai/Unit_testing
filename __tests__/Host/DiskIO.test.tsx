import '../../__mocks__/fakeDependencies'
import React from 'react'
import { BrowserRouter as Router } from 'react-router-dom'
import { mount, render } from 'enzyme'
import DiskIO from '../../pages/Host/DiskIO'

import { render as RTLRender } from '@testing-library/react'
import API from '../../api/Host'

const params = {
    ip: "",
    tag: "",
    measurement: "",
    agg_func: "",
}

describe('<DiskIO />', () => {
    test('expect the DiskIO exist', () => {
        const DiskIOComponent = mount(<Router><DiskIO {...params} /></Router>)
        expect(DiskIOComponent.exists()).toBe(true)
    })

    test('expect DiskIO unmount', () => {
        const DiskIOComponent = mount(<Router><DiskIO {...params} /></Router>)
        DiskIOComponent.unmount()
        expect(() => {
            DiskIOComponent.instance()
        }).toThrow()
    })

    // 快照
    test('the source code changed or not', () => {
        const DiskIOComponent = render(<Router><DiskIO {...params} /></Router>)
        expect(DiskIOComponent).toMatchSnapshot()
    })

    test('detailRegistryHost api should be called if ip has value', () => {
        params.ip = "test"
        const spyFn = jest.spyOn(API, 'diskIORegistryTrend')
        RTLRender(<Router><DiskIO {...params} /></Router>)
        expect(spyFn).toBeCalled()
    })
})