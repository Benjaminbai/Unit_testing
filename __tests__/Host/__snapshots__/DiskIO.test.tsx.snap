// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DiskIO /> the source code changed or not 1`] = `<div />`;
